src/lib.rs

```diff
@@ -265,24 +282,20 @@ where
                     shortcut_text.push_str(&format!("`{}`", text));
                 }
                 if let Some(text_for_header) = state.text_for_header.as_mut() {
-                    let code = format!("{}{}{}", options.code_block_token, text, options.code_block_token);
+                    let code = format!("`{}`", text);
                     text_for_header.push_str(&code);
                 }
-                let (start, end) = if text.contains(options.code_block_token) {
-                    (
-                        String::from_iter([options.code_block_token, options.code_block_token, ' ']),
-                        String::from_iter([' ', options.code_block_token, options.code_block_token]),
-                    )
+                if text.chars().all(|ch| ch == ' ') {
```

Comment:
Would it be wrong to make this more general? It works that way, but that might just mean there aren't enough tests.

```suggestion
                if text.chars().all(|ch| ch.is_whitespace()) {
```

Probably that's also related with the pattern matching later where space is explicitly assumed, but I'd love to see what happens with other whitespace or is the space character indeed special in markdown?

Reply:
Good question! It turns out the [spec has an example about precisely this](https://spec.commonmark.org/0.30/#example-333). It says _Only [spaces](https://spec.commonmark.org/0.30/#space), and not [unicode whitespace](https://spec.commonmark.org/0.30/#unicode-whitespace) in general, are stripped [...]_.